server/blob/blob-objectstorage/src/main/java/org/apache/james/blob/objectstorage/aws/AwsS3ObjectStorage.java

```diff
@@ -132,10 +137,28 @@ private void put(BucketName bucketName, AwsS3AuthConfiguration configuration, Bl
                 .upload(request)
                 .waitForUploadResult();
         } catch (AmazonClientException | InterruptedException e) {
+            if (needToCreateBucket(e)) {
+                createBucket(bucketName, configuration);
+            }
             throw new RuntimeException(e);
```

Comment:
you probably don't want to throw in every cases? And what about retrying the put operation?

Reply:
recursion + max-retry ?